src/main/scala/mesosphere/marathon/api/v2/AppTasksResource.scala

```diff
@@ -29,12 +29,17 @@ class AppTasksResource @Inject() (service: MarathonSchedulerService,
   @GET
   @Produces(Array(MediaType.APPLICATION_JSON))
   @Timed
-  def indexJson(@PathParam("appId") appId: String): Response = {
+  def indexJson(
+    @PathParam("appId") appId: String,
+    @QueryParam("limit")@DefaultValue("-1") limit: Int,
+    @QueryParam("offset")@DefaultValue("0") offset: Int): Response = {
```

Comment:
Wouldn't it be better to validate the parameters at the beginning of the method? offset >= 0, appId non null?